src/arithmetic.c

```diff
@@ -0,0 +1,129 @@
+/**
+ * @file arithmetic.c
+ * @author David Chung (david.chung1719@outlook.com)
+ * @brief Defines the different arithmetic operations used in main.c
+ * @version 3.0
+ * @date 02Dec24
+ * 
+ * @copyright Copyright (c) 2024
+ * 
+ */
+
+#include <stdint.h>
+#include "arithmetic.h"
+#include "debug.h"
+
+int32_t add(int32_t x, int32_t y, int32_t *res)
+{
+    // Check for overflow
+    if ((0 < x) && (0 < y) && ((INT32_MAX - y) < x)) 
+    {
+        fprintf(stderr, "\nOverflow: Result is too large\n");
+        return -1;
```

Comment:
Same thing with a single return point at the end of the function.

```suggestion
int rc = -1; // init as failure

// do stuff, if success condition 
rc = 0;
goto end;

// if failure condition 
goto end;

// at bottom
end:
  return rc;
```